cmd/metal-api/internal/service/switch-service.go

```diff
@@ -166,16 +207,17 @@ func (r switchResource) registerSwitch(request *restful.Request, response *restf
 			return
 		}
 
-		// TODO: Broken switch: A machine was registered before this new switch is getting registered
```

Comment:
I think the comment does not fully relate to the broken switch story due to the way you are solving the issue here. However, it seems the comment still contains a valid point. What happens when there is only one switch registered, a machine registers and after that the second switch comes up? In my opinion we should validate the two-neighbor condition when a machine registers and also when a machine gets allocated, which relates to https://github.com/metal-stack/metal-api/issues/21.

Maybe just remove "Broken switch" from the comment?

Reply:
Yes, this is an issue. We now assume stronger constraints during the switch replacement then for the machine registration!

I would suggest to add these constraints to the machine registration:
* there must be two distinct switches that have connections to the machine
* these two switches must have "equal" machineConnections with respect to the switch port names

This would also help with #21

Reply:
I added operational steps needed to replace the switch in https://github.com/metal-stack/metal-api/issues/30

Reply:
Sounds good.
Regarding operational steps... how do I put the switch into replace mode as an operator? Will it become a metalctl command?

Reply:
Just added the additional constraints for the machine registration as well.

`metalctl switch apply / edit <ID>` is probably sufficient to put a switch into `replace` mode or back to normal `operational` mode.

A flag like `--mode` is not that appropriate as we just have those kind of flags for creation of entities but, as of now, not for editing.